### PR TITLE
[PORT] Crew monitor new icons (READY)

### PR DESCRIPTION
**Pull request by warface1234455**
# About PR
Ports:
- https://github.com/Skyrat-SS13/Skyrat-tg/pull/12744
- https://github.com/Skyrat-SS13/Skyrat-tg/pull/12147

# Document the changes in your pull request
Tweaked colors a bit and added some icons
New health icon status and race icons
Any other races will showed as red mark
Suit sensors now check your species and if you are on fire, irradiated, wounded, husked

# Updated images
184630902-349cd6c9-a6d4-49fb-8f19-41cecbb10215

image
from left to right: irradiated(500 rad and above), husked(If reaches 241 burn damage), on fire, wounded(All wounds), Bone wound(only bone wound will appear here), species type, vitals, location

# Wiki Documentation
Crew monitor has new health status and race icons
# Changelog

:cl:  [Zergspower](https://github.com/Zergspower), [Deek-Za](https://github.com/Deek-Za) ported by warface1234455 (Also added a few things)
rscadd: suit sensor now check your race and if you have wound, husk, on fire and irradiation
imageadd: replaced old color box icons with new health status icons for crew monitor
imageadd: added some health check icons for wound, husk, on fire and irradiation
imageadd: added some race icons
rscadd: Added issnail for is_helpers
/:cl:

### CODE REVIEW

**Comment by warface1234455** [comment]
why lints failed aaa imma go rice and figure this later

**Comment by KillerOrcaCora** [comment]
White forgor skeleton because you really are just gonna be forgotten

**Comment by warface1234455** [comment]
> White forgor skeleton because you really are just gonna be forgotten

What

**Comment by ghost** [comment]
> why lints failed aaa imma go rice and figure this later

/home/runner/work/Yogstation/Yogstation/tgui/packages/tgui/interfaces/CrewConsole.js
    2:49  error  A space is required before '}'                            object-curly-spacing
   57:1   error  This line has a length of 104. Maximum allowed is [100](https://github.com/yogstation13/Yogstation/runs/7817999257?check_suite_focus=true#step:7:104)     max-len
  152:1   error  Expected indentation of 18 spaces but found 16            indent
  152:17  error  Expected indentation of 18 space characters but found 16  react/jsx-indent
  153:1   error  Expected indentation of 20 spaces but found 18            indent
  154:1   error  Expected indentation of 22 spaces but found 20            indent
  155:1   error  Expected indentation of 24 spaces but found 22            indent
  156:1   error  Expected indentation of 24 spaces but found 22            indent
  157:1   error  Expected indentation of 24 spaces but found 22            indent
  158:1   error  Expected indentation of 24 spaces but found 22            indent
  159:1   error  Expected indentation of 24 spaces but found 22            indent
  160:1   error  Expected indentation of 24 spaces but found 22            indent
  161:1   error  Expected indentation of 22 spaces but found 20            indent
  162:1   error  Expected indentation of 24 spaces but found 22            indent
  163:1   error  Expected indentation of 24 spaces but found 22            indent
  164:1   error  Expected indentation of 24 spaces but found 22            indent
  165:1   error  Expected indentation of 24 spaces but found 22            indent
  166:1   error  Expected indentation of 24 spaces but found 22            indent
  167:1   error  Expected indentation of 24 spaces but found 22            indent
  168:1   error  Expected indentation of 22 spaces but found 20            indent
  169:1   error  Expected indentation of 18 spaces but found 16            indent
  170:1   error  Expected indentation of 20 spaces but found 18            indent
  171:1   error  Expected indentation of 22 spaces but found 20            indent
  172:1   error  Expected indentation of 20 spaces but found 18            indent
  173:1   error  Expected indentation of 22 spaces but found 20            indent
  174:1   error  Expected indentation of 20 spaces but found 16            indent

✖ 26 problems (26 errors, 0 warnings)
  25 errors and 0 warnings potentially fixable with the `--fix` option.

**Comment by warface1234455** [comment]
I can see

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 1]
```diff
@@ -140,13 +149,29 @@ export const CrewConsoleContent = (props, context) => {
                   ({!originalTitles ? sensor.assignment_title : sensor.assignment})
```
Comment (warface1234455) [inline tgui/packages/tgui/interfaces/CrewConsole.js:149]: hm

**Comment by warface1234455** [comment]
ok fixing lints broke the computer

**Comment by warface1234455** [comment]
my life is a pain

**Comment by warface1234455** [comment]
imma add some race icons

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 2]
```diff
@@ -140,13 +154,64 @@ export const CrewConsoleContent = (props, context) => {
...
-                      sensor.burndam,
-                      sensor.brutedam,
-                      sensor.life_status)} />
+                  {sensor.is_irradiated ? <Icon name="radiation" color="#f0e21d" size={1} /> : ""}
+                  {sensor.is_husked ? <Icon name="ribbon" color="#ad1c09" size={1} /> : ""}
+                  {sensor.is_onfire ? <Icon name="fire" color="#f24f0f" size={1} /> : ""}
+                  {sensor.is_wounded ? <Icon name="star-of-life" color="#f50505" size={1} /> : ""}
+                </Table.Cell>
+                <Table.Cell collapsing textAlign="center">
+                  {sensor.is_robot ? (<Icon name="robot" color="#2e46cc" size={1} />
```
Comment (redmoogle) [inline tgui/packages/tgui/interfaces/CrewConsole.js]: holy shit my eyes. you could pull the `species.name` and have a mapping to color such as

```
ipc = "#FFFFFF",
felinid = "#000000"

**Comment by JamieD1** [comment]
Is there no way to make the map work again?

**Comment by warface1234455** [comment]
what

File: tgui/packages/tgui/interfaces/CrewConsole.js (hunk 2 above)
Comment (ShoziOffical) [reply tgui/packages/tgui/interfaces/CrewConsole.js]: didnt know about that.. oh well

**Comment by warface1234455** [comment]
> Is there no way to make the map work again?

wdym

**Comment by JamieD1** [comment]
> Is there no way to make the map work again?

^

**Comment by ShoziOffical** [comment]
idk anything about the old map, i didnt even get to see it

**Review by Bibby0110 — DISMISSED** [review]
Code quality poopoo, review soon

**Comment by ShoziOffical** [comment]
> Code quality poopoo, review soon

i have less knowledge in coding so im basically just porting stuffs but i do test it to make sure everything works

**Comment by Bibby0110** [comment]
> > Code quality poopoo, review soon
> 
> i have less knowledge in coding so im basically just porting stuffs but i do test it to make sure everything works

it is ok, that is why we have maintainers

**Comment by JamieD1** [comment]
@TheGamerdk Get on it please.

**Review by JamieD1 — APPROVED** [review]

**Review by redmoogle — CHANGES_REQUESTED** [review]
for the love of god fix the code

**Comment by JamieD1** [comment]
> for the love of god fix the code
Can you suggest fixes?

**Comment by redmoogle** [comment]
> > for the love of god fix the code
> > Can you suggest fixes?

see my initial review comment

**Review by JamieD1 — DISMISSED** [review]
Please look at moogles review

**Comment by ShoziOffical** [comment]
~~Im in hospital probably gonna stay there for a week~~ got discharged currently working on it

**Comment by ShoziOffical** [comment]
done @JamieD1

File: code/game/machinery/computer/crew.dm [hunk 3]
```diff
@@ -161,7 +166,56 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
...
+				if (isjellyperson(H))
+					species = "Slime"
+				if (isethereal(H))
+					species = "Ethereal"
+				if (iszombie(H))
+					species = "Zombie"
+				if (issnail(H))
+					species = "Snail"
 
+				if (H.radiation > 500) //safe level before sending alert
```
Comment (redmoogle) [inline code/game/machinery/computer/crew.dm]:
```suggestion
				if (H.radiation > RAD_MOB_SAFE) //safe level before sending alert
```

**Review by redmoogle — APPROVED** [review]
Code looks fine
could be improved by doing var/x = FALSE instead of setting it in a else statement

**Comment by ShoziOffical** [comment]
@TheGamerdk review

**Comment by ShoziOffical** [comment]
ok lemme fix the thing CHIULL

File: code/__DEFINES/is_helpers.dm [hunk 4]
```diff
@@ -81,6 +81,7 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 #define ispreternis(A) (is_species(A,/datum/species/preternis))
 #define isszlachta(A) (is_species(A, /datum/species/szlachta))
 #define isipc(A) (is_species(A, /datum/species/ipc))
+#define issnail(A) (is_species(A, /datum/species/snail))
```
Comment (warface1234455) [inline code/__DEFINES/is_helpers.dm:84]: AAAAA

**Comment by ShoziOffical** [comment]
done and done pls review :(

**Review by KillerOrcaCora — APPROVED** [review]

**Comment by ShoziOffical** [comment]
what you can't request two people?? brah github, since jamie is on break sooo

**Pull request merged**